### ISSUE TITLE
Language-Map definition needs to be loosened.

### ISSUE THREAD

**Issue by deathtenk** [yetanalytics/xapi-schema#65]
So this came out of a back and fourth between us and Valamis. A statement that they consider valid came through with a language map as an empty object. Our validator handles it like so:

```clojure
{:cljs.spec.alpha/problems
 ({:path
   [:statement/object
    :statement-object/activity
    :activity/definition
    nil
    :definition/description],
   :pred (cljs.core/<= 1 (cljs.core/count %) 9007199254740991),
   :val {},
   :via
   [:xapi-schema.spec/statement
    :statement/object
    :xapi-schema.spec/activity
    :activity/definition
    :xapi-schema.spec/language-map],
   :in
   [:statement/object :activity/definition :definition/description]}),
 :cljs.spec.alpha/spec :xapi-schema.spec/statement,
 :cljs.spec.alpha/value
 {"object"
  {"id"
   "https://localhost/valamis/lessons/352c53fe-afea-49dd-a3c9-971cfd564c8a",
   "objectType" "Activity",
   "definition"
   {"name" {"en-US" "Cat Pictures"},
    "description" {},
    "type" "http://adlnet.gov/expapi/activities/course"}},
  "authority"
  {"objectType" "Group",
   "member"
   [{"objectType" "Agent",
     "account"
     {"homePage" "http://cpaodev-lrs:8080/xapi/OAuth/Token",
      "name" "9ca5b765-a084-4a72-b801-cb8500c2861e"}}
    {"objectType" "Agent",
     "account"
     {"homePage" "http://localhost",
      "name" "9ca5b765-a084-4a72-b801-cb8500c2861e"}}]},
  "verb"
  {"id" "http://adlnet.gov/expapi/verbs/completed",
   "display" {"en-US" "completed"}},
  "id" "cdf843cb-7f5e-4ac4-a4ca-583fdd5f40e5",
  "timestamp" "2019-06-17T18:58:40.215Z",
  "context"
  {"registration" "3dc57648-0854-4ca7-bd90-95ee396c7ad2",
   "contextActivities"
   {"parent"
    [{"id"
      "https://localhost/valamis/lessons/352c53fe-afea-49dd-a3c9-971cfd564c8a",
      "objectType" "Activity"}]},
   "revision" "version 1.1"},
  "version" "1.0.0",
  "stored" "2019-06-17T18:59:39.990Z",
  "actor"
  {"objectType" "Agent",
   "account"
   {"homePage" "http://localhost",
    "name" "29f36577-104a-58c6-55d8-90c9ab5c168e"},
   "name" "29f36577-104a-58c6-55d8-90c9ab5c168e"},
  "result"
  {"score" {"scaled" 1}, "success" true, "duration" "PT92H32M50S"}}}
```
However, the spec doesn't explicitly say that you MUST disclude empty objects, only SHOULD. (source: https://github.com/adlnet/xAPI-Spec/blob/master/xAPI-Data.md#22-formatting-requirements) . In regard to Language Maps, it talks about the addition of language tags for different entries being enforced, but there's no rule that disallows language-maps from being empty. 

Because of this, the validator in Valamis' LRS and also ADL's validator allows empty objects as language maps (you can run a statement with an empty language map through ADLs validator to see this in action: https://lrs.adlnet.gov/statementvalidator). Given the way the spec is worded and the way that other systems work we need to line up our validator to reflect it. The consequence of not following it as interpreted by ADL is we won't be able to line up our integrations with other systems that interpret the rules the same way.

**Comment by kelvinqian00** [comment]
Fixed in PR #71

**Comment by kelvinqian00** [comment]
After a long Slack discussion, we have decided that this issue is not worth fixing, for a number of reasons:
- Despite empty language maps being technically allowed under the xAPI spec, in practice they should not happen, due to being non-sensical.
- Some apps that use xapi-schema as a dependency cannot accept data with empty maps, e.g. `nave` and `third` use DynamoDB, which does not accept empty maps.
- Relating to the above, most systems will/should fail when an empty language map is encountered. Valamis is the exception to the rule, not the standard. If a similar exception comes up in the future, it's better to add custom middleware instead of potentially breaking most of our other apps.
